src/odemis/acq/align/test/fastem_test.py

```diff
@@ -0,0 +1,185 @@
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the terms
+of the GNU General Public License version 2 as published by the Free Software
+Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+
+"""
+from __future__ import division
+
+import logging
+import os
+import time
+import unittest
+from concurrent.futures._base import CancelledError
+
+import numpy
+
+import odemis
+from odemis import model
+from odemis.acq.align import fastem
+from odemis.acq.align.fastem import OPTICAL_AUTOFOCUS, IMAGE_TRANSLATION_PREALIGN
+from odemis.util import test
+
+# * TEST_NOHW = 1: use simulator (asm/sam and xt adapter simulators need to be running)
+# * TEST_NOHW = 0: connected to the real hardware (backend needs to be running)
+# technolution_asm_simulator/simulator2/run_the_simulator.py
+TEST_NOHW = (os.environ.get("TEST_NOHW", "0") != "0")  # Default is HW testing
+
+logging.getLogger().setLevel(logging.DEBUG)
+logging.basicConfig(format="%(asctime)s  %(levelname)-7s %(module)s:%(lineno)d %(message)s")
+
+CONFIG_PATH = os.path.dirname(odemis.__file__) + "/../../install/linux/usr/share/odemis/"
+FASTEM_CONFIG = CONFIG_PATH + "sim/fastem-sim-asm.odm.yaml"
+
+
+class TestFastEMCalibration(unittest.TestCase):
+    """Test the calibration manager."""
+
+    @classmethod
+    def setUpClass(cls):
+        if TEST_NOHW:
+            test.start_backend(FASTEM_CONFIG)
+
+        # get the hardware components
+        cls.scanner = model.getComponent(role='e-beam')
+        cls.asm = model.getComponent(role="asm")
+        cls.mppc = model.getComponent(role="mppc")
+        cls.multibeam = model.getComponent(role="multibeam")
+        cls.descanner = model.getComponent(role="descanner")
+        cls.stage = model.getComponent(
+            role="stage")  # TODO replace with stage-scan when ROA conversion method available
+        cls.ccd = model.getComponent(role="diagnostic-ccd")
+        cls.focuser = model.getComponent(role="diagnostic-cam-focus")
+        cls.beamshift = model.getComponent(role="ebeam-shift")
+        cls.det_rotator = model.getComponent(role="det-rotator")
+
+    def setUp(self):
+
+        self.good_focus = -70e-6  # position where the image of the multiprobe is displayed in focus [m]
+        self.good_mp_position = (650, 450)  # position for multiprobe where mp is aligned with mppc detector [px_dc]
+        self.ccd.updateMetadata(
+            {model.MD_FAV_POS_ACTIVE:
+                {"x": self.good_mp_position[0],
+                 "y": self.good_mp_position[1],
+                 "z": self.good_focus}}
+                                )
```

Comment:
Since you're not using `self.good_mp_position` anywhere, if you don't want to explicitly set it and just use the values from the yaml file, you could update these lines to:
```
self.ccd.updateMetadata(model.MD_FAV_POS_ACTIVE.update({"z": self.good_focus})
```

Reply:
True. Maybe in future we can test the image translation pre-alignment in a fancier way, where we want to use explicit values. I removed it for now.